dbms/src/Operators/LocalAggregateTransform.cpp

```diff
@@ -42,24 +44,55 @@ OperatorStatus LocalAggregateTransform::transformImpl(Block & block)
     switch (status)
     {
     case LocalAggStatus::build:
-        if (unlikely(!block))
+        if unlikely (!block)
         {
-            // status from build to convert.
-            status = LocalAggStatus::convert;
-            agg_context.initConvergent();
-            if likely (!agg_context.useNullSource())
-            {
-                RUNTIME_CHECK(agg_context.getConvergentConcurrency() == local_concurrency);
-                block = agg_context.readForConvergent(task_index);
-            }
-            return OperatorStatus::HAS_OUTPUT;
+            return agg_context.hasSpilledData()
+                ? fromBuildToFinalSpillOrRestore()
+                : fromBuildToConvergent(block);
         }
         agg_context.buildOnBlock(task_index, block);
         block.clear();
```

Comment:
Why we need to clear the block?

Reply:
Just to release the memory earlier, the block written to the sink will not be used anymore